args.go

```diff
@@ -44,28 +45,85 @@ func (cfg *Config) UpgradeDir(upgradeName string) string {
 	return filepath.Join(cfg.Root(), upgradesDir, safeName)
 }
 
+//Symbolinking to genesis
+func (cfg *Config) SymLinkToGenesis() string  {
+	genesisBin := cfg.GenesisBin()
+
+	info, err := os.Lstat(genesisBin)
+	if err != nil {
+		return ""
+	}
+
+	// if it is there, ensure it is a symlink
+	if info.Mode()&os.ModeSymlink == 0 {
+		return ""
+	}
+
+	// resolve it
+	dest, err := os.Readlink(genesisBin)
+	if err != nil {
+		return ""
+	}
+
+	genBinary := filepath.Join(dest, "bin", cfg.Name)
+
+	//Write current binary to file
+	WriteToFile(genBinary)
+
+	// and return the genesis binary
+	return genBinary
+}
+
 // CurrentBin is the path to the currently selected binary (genesis if no link is set)
 // This will resolve the symlink to the underlying directory to make it easier to debug
 func (cfg *Config) CurrentBin() string {
 	cur := filepath.Join(cfg.Root(), currentLink)
 	// if nothing here, fallback to genesis
 	info, err := os.Lstat(cur)
 	if err != nil {
-		return cfg.GenesisBin()
+		//Create symbolink to the genesis
+		return cfg.SymLinkToGenesis()
```

Comment:
Yes, replace `cfg.GenesisBin()` => `cfg.SymlinkToGenesis()` is a good change. 
The parts at the end not really necessary